lib/enkf/row_scaling.cpp

```diff
@@ -21,12 +21,152 @@
 #include <stdexcept>
 #include <vector>
 
-
 #include <ert/enkf/row_scaling.hpp>
+#include <ert/res_util/matrix.hpp>
+#include <ert/util/util.hpp>
+
+#define ROW_SCALING_ID 6123199
+#define ROW_SCALING_RESOLUTION 1000
+
+
+/*
+  The values in the row_scaling container are distributed among
+  ROW_SCALING_RESOLUTION discreete values. The reason for this lumping is to be
+  able to reuse the scaled update matrix:
+
+      X(a) = a*X0 + (1 - a)*I
+
+  for several rows. This grouping is utilized in the row_scaling_multiply()
+  function.
+*/
+
+
+namespace {
+
+void scaleX(matrix_type* X, const matrix_type * X0, double alpha) {
+  matrix_assign(X, X0);
+  matrix_scale(X, alpha);
+  for (int i=0; i < matrix_get_rows(X); i++)
+    matrix_iset(X, i, i, (1 - alpha) + matrix_iget(X, i, i));
+}
+
+}
+
+int row_scaling::size() const {
+  return this->data.size();
+}
+
+
+double row_scaling::operator[](int index) const {
+  return this->data.at(index);
+}
+
+void row_scaling::assign(int index, double value) {
+  if (value < 0 || value > 1)
+    throw std::invalid_argument("Invalid value ");
+
+  if (this->data.size() <= index)
+    this->data.resize(index + 1, 1);
```

Comment:
This automatic resizing is a bit questionable.  While it is convenient, a more explicit usage could prevent hard to discover side effects